docs/Guides/V2Upgrade.md

```diff
@@ -8,36 +8,86 @@ Account V1’s single EOA ownership is comparable to Account V2 with the ECDSA O
 
 In the [SDK release](https://github.com/bcnmy/biconomy-client-sdk/releases/tag/r6) (npm version 3.1.1), below API methods have been added to V1 Account API specs (BiconomySmartAccount.ts in accounts package) to help with the upgrades
 
-- **getUpdateImplementationData( )** : This returns transaction object which can be used in buildUserOp() method for batching with other transactions.
+- **getUpdateImplementationData( )** : This returns transaction object which can be used in buildUserOp() method for batching with other transactions. To use this method one must enable a validation module. This method can be used while batching the upgrade with other transactions.
 
-- **updateImplementationUserOp( )** : This returns the userOp, which can be sent as a single action to do the upgrade.
+- **updateImplementationUserOp( )** : This returns the userOp, which can be sent as a **single** action to do the upgrade. 
 
 ### Steps to upgrade
 
-1. create a Biconomy account V1 instance and execute **`getUpdateImplementationData`** transaction.
+1. create a Biconomy account V1 instance.
 
 ```tsx
   const biconomyAccount = new BiconomySmartAccount(biconomySmartAccountConfig);
   const biconomySmartAccount = await biconomyAccount.init();
+```
 
-  const tx = await biconomySmartAccount.getUpdateImplementationData() 
-  console.log('account update tx', tx)
-
-  // transaction is any existing transaction
-
-  let partialUserOp = await biconomySmartAccount.buildUserOp([transaction, tx]);
+2. Create partial userOp using one of the following methods depending on the usecase. 
+- If user wants to batch the upgrade transaction with other transactions, `getUpdateImplementationData` method can be used along with `getModuleSetupData`. 
+
+:::info
+Its critical to add the **moduleSetupData** while doing the upgrade when using getUpdateImplementationData to ensure atleast one validation method can be enabled.
```

Comment:
```suggestion
Its critical to add the **moduleSetupData** while doing the upgrade when using getUpdateImplementationData to ensure at least one validation module is enabled. Otherwise your V2 account will be left without any authorization and unusable.
```